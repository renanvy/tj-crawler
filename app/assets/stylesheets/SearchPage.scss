.SearchPage {
  &_title {
    font-size: 24px;
    font-weight: 500;
  }
}