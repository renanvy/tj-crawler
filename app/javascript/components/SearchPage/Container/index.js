import React from 'react'

export default function Container(props) {
  return (
    <div className="card">
      <div className="card-content">
        <span className="card-title">Processo n. 234354.343 do TJAL</span>

        <span>Distribuido em 09/11/19</span>
        
        <div className="row">
          {props.children}
        </div>
      </div>
    </div>
  )
}