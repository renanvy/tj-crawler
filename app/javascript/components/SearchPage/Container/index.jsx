import React, { Component } from "react";
import moment from "moment";
import PropTypes from "prop-types";

import "./styles.scss";

import Preloader from "../../common/Preloader";

export default class Container extends Component {
  renderContent() {
    const { process, loading, children } = this.props;

    if (loading) {
      return <Preloader />;
    }

    if (!process) {
      return <p className="center">Nenhum processo encontrado.</p>;
    }

    return (
      <>
        <h1 className="Container_title">
          Processo nº.
          {process.number}
          do TJAL
        </h1>

        <span className="Container_crawler_updated_at">
          Última atualização em          {" "}
          {moment(process.crawler_updated_at).format("DD/MM/YYYY HH:mm")}
        </span>

        <div className="row">{children}</div>
      </>
    );
  }

  render() {
    return (
      <div className="card">
        <div className="card-content">{this.renderContent()}</div>
      </div>
    );
  }
}

Container.propTypes = {
  process: PropTypes.object,
  loading: PropTypes.bool.isRequired,
  children: PropTypes.node
};
