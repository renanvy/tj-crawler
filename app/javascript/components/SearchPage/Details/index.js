import React from "react";
import PropTypes from "prop-types";

import "./styles.scss";

export default function Details(props) {
  const { process } = props;
  return (
    <div className="col s12 m4">
      <h1 className="Details_title">Detalhes do processo</h1>

      <ul className="Details_list_infos">
        <li>
          <strong>Distribuído em:</strong>
          {process.distribution_date}
        </li>
        <li>
          <strong>Classe: </strong>
          {process.kind}
        </li>
        <li>
          <strong>Área: </strong>
          {process.area}
        </li>
        <li>
          <strong>Assunto: </strong>
          {process.subject}
        </li>
        <li>
          <strong>Juíz: </strong>
          {process.judge}
        </li>
        <li>
          <strong>Valor da Ação: </strong>
          {process.action_value}
        </li>
      </ul>

      <div className="divider" />

      <br />

      <h1 className="Details_title">Partes envolvida</h1>

      <ul>
        <li>
          <a href="!#">Nelson Willians & Advogados Associados</a>
          <br />
          <span>Parte envolvida * Exequente</span>
        </li>

        <li>
          <a href="!#">Nelson Willians & Advogados Associados</a>
          <br />
          <span>Parte envolvida * Exequente</span>
        </li>

        <li>
          <a href="!#">Nelson Willians & Advogados Associados</a>
          <br />
          <span>Parte envolvida * Exequente</span>
        </li>

        <li>
          <a href="!#">Nelson Willians & Advogados Associados</a>
          <br />
          <span>Parte envolvida * Exequente</span>
        </li>
      </ul>
    </div>
  );
}

Details.propTypes = {
  process: PropTypes.object
};
