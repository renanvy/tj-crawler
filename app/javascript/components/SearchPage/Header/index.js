import React, { Component } from "react";
import PropTypes from "prop-types";

export default class Header extends Component {
  constructor(props) {
    super(props);

    this.state = {
      processNumber: ""
    };
  }

  componentDidMount() {
    this.startsMaterializeSelect();
  }

  startsMaterializeSelect() {
    const materializeSelect = document.querySelectorAll("select");
    M.FormSelect.init(materializeSelect, {});
  }

  render() {
    const { loading, search } = this.props;
    const { processNumber } = this.state;

    return (
      <header>
        <form onSubmit={evt => search(evt, processNumber)}>
          <div className="row">
            <div className="input-field col s12 m3">
              <select defaultValue="tjal">
                <option value="tjal" disabled>
                  TJAL
                </option>
              </select>
              <label>Tribunal</label>
            </div>

            <div className="input-field col s12 m5">
              <input
                id="process_number"
                type="text"
                className="validate"
                value={processNumber}
                onChange={evt =>
                  this.setState({
                    processNumber: evt.target.value
                  })
                }
              />

              <label htmlFor="process_number">NÃºmero do Processo</label>
            </div>

            <button
              style={{ marginTop: 5 }}
              type="submit"
              className="btn btn-large"
              disabled={loading || !processNumber}
            >
              <i className="material-icons">search</i>
            </button>
          </div>
        </form>
      </header>
    );
  }
}

Header.propTypes = {
  loading: PropTypes.bool.isRequired,
  search: PropTypes.func.isRequired
};
