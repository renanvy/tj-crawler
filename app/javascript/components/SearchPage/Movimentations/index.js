import React from "react";
import moment from "moment";
import PropTypes from "prop-types";

export default function Movimentations(props) {
  const { movimentations } = props;

  return (
    <div className="Movimentations">
      <div className="col s12 m8">
        <ul className="collection with-header">
          <li className="collection-header">
            <h1 className="Movimentations_title">Movimentações</h1>
          </li>

          {movimentations.map(movimentation => (
            <li
              key={`movimentation-${movimentation.id}`}
              className="collection-item"
            >
              <span className="Movimentation_date">
                {moment(movimentation.date).format("DD/MM/YYYY")}
              </span>

              <p className="Movimentation_description">
                {movimentation.description}
              </p>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
}

Movimentations.propTypes = {
  movimentations: PropTypes.array.isRequired
};
