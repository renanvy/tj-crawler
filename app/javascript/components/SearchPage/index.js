import React, { Component } from "react";

import { routes } from "../../api";

import Header from "./Header";
import Container from "./Container";
import Movimentations from "./Movimentations";
import Details from "./Details";

export default class SearchPage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      process: null,
      movimentations: [],
      loading: false
    };

    this.search = this.search.bind(this);
  }

  async search(evt, processNumber) {
    evt.preventDefault();

    this.setState({ loading: true, process: null, movimentations: [] });

    try {
      const response = await fetch(
        `${routes.processes_path}?number=${processNumber}`
      );

      const process = await response.json();
      const { movimentations } = process;

      this.setState({ process, movimentations });
    } catch (e) {
      M.toast({
        html: "Ocorreu um erro ao carregar o processo. Tente novamente."
      });
    } finally {
      this.setState({ loading: false });
    }
  }

  render() {
    const { process, movimentations, loading } = this.state;

    return (
      <div className="container">
        <h1 className="SearchPage_title">Consulta de processos de 1º grau.</h1>

        <blockquote>
          Informe no campo de busca o nº do processo que deseja buscar no TJAL
          (Tribunal de Justiça do Estado de Alagoas).
        </blockquote>

        <Header search={this.search} loading={loading} />

        <Container process={process} loading={loading}>
          <Movimentations movimentations={movimentations} />
          <Details process={process} />
        </Container>
      </div>
    );
  }
}
