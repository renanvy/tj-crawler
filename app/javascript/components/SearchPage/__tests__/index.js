import React from "react";
import renderer from "react-test-renderer";
import SearchPage from "../index";
import "materialize-css/dist/js/materialize.min";

describe("SearchPage", () => {
  test("snapshot renders", () => {
    const component = renderer.create(<SearchPage />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
