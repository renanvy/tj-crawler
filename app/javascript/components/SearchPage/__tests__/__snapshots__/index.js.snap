// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage snapshot renders 1`] = `
<div
  className="container"
>
  <h1
    className="SearchPage_title"
  >
    Consulta de processos de 1º grau.
  </h1>
  <blockquote>
    Informe no campo de busca o nº do processo que deseja buscar no TJAL (Tribunal de Justiça do Estado de Alagoas).
  </blockquote>
  <header>
    <form
      onSubmit={[Function]}
    >
      <div
        className="row"
      >
        <div
          className="input-field col s12 m3"
        >
          <select
            defaultValue="tjal"
          >
            <option
              disabled={true}
              value="tjal"
            >
              TJAL
            </option>
          </select>
          <label>
            Tribunal
          </label>
        </div>
        <div
          className="input-field col s12 m5"
        >
          <input
            className="validate"
            id="process_number"
            onChange={[Function]}
            type="text"
            value=""
          />
          <label
            htmlFor="process_number"
          >
            Número do Processo
          </label>
        </div>
        <button
          className="btn btn-large"
          disabled={true}
          type="submit"
        >
          <i
            className="material-icons"
          >
            search
          </i>
        </button>
      </div>
    </form>
  </header>
  <div
    className="card"
  >
    <div
      className="card-content"
    >
      <p
        className="center"
      >
        Nenhum processo encontrado.
      </p>
    </div>
  </div>
</div>
`;
